### DISCUSSION TITLE
What Does the Future of Coding Look Like in the Age of AI?

### DISCUSSION THREAD

**Discussion by mwzkhalil** [codewars/codewars.com#3018] [General]
I'm intrigued by the future of coding in the age of AI. With rapid advancements in ai, what transformations can we anticipate in the coding landscape? How will AI impact coding practices, job prospects, and the overall industry?

**Comment by nosrednarecneps** [comment]
I feel like I have a decent idea in where web developers in particular may head to in the future. With advancements in AI, LowCode-NoCode tools could become even more prevalent as these tools could begin relying on AI to develop complex elements within a webpage. The future for web developers is interesting as I believe web developers may begin to specialize in more of the design of a website as technical expertise gets more and more automated by NoCode tools and Ai. My prediction is in the next 10-20 years, the role of web developers may change. Other types of developers may also see Jurassic changes in the way they develop, but in my opinion, web developers may see the most change.

**Reply by aryuemaan** [reply]
I feel Like more replacement will come but, they used to say calculators will replace mathematicians but it couldn't 
we still need logic builders and forever !

**Reply by LeafBorn** [reply]
It start happening. Your prediction is so right, What is your next perdiction?

**Reply by jenilcodes11** [reply]
And those who are saying coding has no future but you cannot write word without learning alphabet

**Reply by Cyuzuzofabrice** [reply]
I am junior developer i am looking for internaship
###

**Comment by Amrit-ghale** [comment]
The future of coding in the age of AI holds promise for increased productivity, enhanced collaboration, and more accessible programming. AI will augment developers' capabilities, automate mundane tasks, and facilitate smarter coding practices. It is important to recognize that while AI will bring advancements, human creativity, problem-solving, and critical thinking will remain essential in software development.

**Reply by DavidovichVA** [reply]
Is that a ChatGPT answer?

**Reply by jroycodes** [reply]
😂😂

**Reply by Mr-Mens** [reply]
lol

**Reply by unclejoefx** [reply]
OMG.  hahahahh

**Reply by leobrines** [reply]
JAJAJJA kjj

**Reply by alex-morgun** [reply]
Good Robot!

**Reply by anjumhere** [reply]
bot spotted 😂

**Reply by mosa-150505** [reply]
It's just for productivity according to me but if u want a clever code, no one can beat human ideas

**Comment by trev-sykes** [comment]
Take a look at the implementations of AI as tutors for the CS50 curriculum. David Malan spoke on the news about how they are integrating AI into their courses. AI, when used in such ways, can be extremely helpful for aspiring developers. There is a fine line between using AI as a tool in your training wheels toolkit and over relying on AI in every aspect of your journey. What the cs50 team is going i applaud. They're, for lack of a better term, dumbing down chat GPT so that it isn't overly helpful and will not give straight answers. CS50 team is training AI how to be a good teacher and guide you towards the right answer rather than give it to you in a lengthy three paragraph answer. I think it is useful for the knowledgable developer who can utilize AI to fine tune their abilities and automate things better. I think it will be the downfall for many aspiring developers who will get into the all too easy habit of opening a new tab, and typing chatGPT in the browser and asking it for help. One of the most important aspects of being a developer is being able to problem solve by researching topics and letting your brain work for the answer. GPT side swipes that entire process and turns what would have been a valuable potential hour of getting to the answer the best way, through trial and error and google searches on methods etc, into a one minute (very informative one minute) search with the answer overly-complicatedly given to you in extraneous detail. I can only speak my thoughts, what do you think about this view?

**Reply by Erikbeute** [reply]
Agreed on this one. It can be a very helpful tool for learning and for syntax-checking and finding just 'dumb' mistakes that everyone overlooks once in a while.
But if your 'learning' or coding consists of copy-pasting from ol' GPT, you won't learn anything.

**Reply by hmenorjr** [reply]
Agree as well. I'm a back-end developer with JavaEE as core. One of the batch jobs that we migrated from Java and Spring Boot to TypeScript and Node.js. However, since I'm not familiar with the tech in total, ChatGPT (with Google) helped me on majority of the code snippets. But I can't run the code locally and use the testing framework that the team used (who's more familiar with the tech). So, I had to rely on console.log(). HEHE.

**Reply by uduma-sonia** [reply]
I agree and this is why I stopped using ChatGPT. I found out that searching for answers on google leads me to materials and resources and I gain additional info during my course of search whereas while using ChatGPT i'm limited to whatever it tells me.

**Comment by somatract** [comment]
AI wont be taking over humans. AI still struggles now with basic coding corrections for syntax. If anything AI will just become another "plug-in" to help out

**Comment by Stelladario** [comment]
The future of coding in the age of AI is poised for transformative changes. As AI tools become more sophisticated, they will not only assist developers but also automate routine tasks, allowing coders to focus on more creative and complex problems. This shift could democratize programming, making it accessible to non-coders through intuitive interfaces. However, a solid understanding of coding principles will remain essential. For students navigating this landscape, [coursework help](https://myassignmenthelp.expert/coursework/) can be invaluable, providing the support needed to grasp foundational concepts while adapting to new technologies. Embracing this evolution will be key to thriving in the AI-driven tech world.

**Comment by BlackDereker** [comment]
AI still struggles to develop complex logic and it's limited context length makes it impossible to take into account all the code in a big project and other services that are being used.

The first thing that I'm seeing disappear is the need of memorizing syntax and that's a good one in my opinion. Developers will focus more on developing an idea/product without caring much what is the specific syntax to do it. People will start learning about how computers work fundamentally instead of just learning how to code.

**Reply by trev-sykes** [reply]
Great insight! I completely agree.  Although Tools like ChatGPT and Claude.ai have made huge strides in prompt recognition, allowing them to infer a lot from just a few lines of code. As we get better at crafting prompts, we can really maximize their effectiveness as code generators, letting developers focus more on the big picture of system design. This shift could lead to more creative and advanced approaches to building software and hardware, while AI handles the details.

**Reply by JAY-cloudbuster** [reply]
trust me, the first two lines helped me to sleep

**Comment by TorkelV** [comment]
Gemini 3.0 can now solve many Kyu 1 problems on Codewars.. Problems that I thought LLM's would never ever solve. Square sums, game of life in Reverse, 7x7 skyscrapers to name a few.
 It can even one shot code golfing problems like "One line task: Is the king in check?"

I'm not sure what to say about that..

**Reply by Madjosz** [reply]
You have to look at it like this: You would also be able to solve a 1 kyu Codewars problem if you just copy the solution from a Github repo. Especially if you learned all of Github by heart.

**Comment by umarcoder5** [comment]
Well, in my opinion, coding will not be a profession anymore because AI will replace humans. However, there should be some people using the AI too, you know what I mean; it's like AI can't just work by itself, can it?

**Reply by bless-rukundo** [reply]
This really true

**Reply by mosa-150505** [reply]
I think there'll be new professions about maintaning AI's code.

**Reply by BiTech5** [reply]
It's true that AI will replace in terms of coding jobs. But companies are still looking for good engineers for reviewing, debugging, refactoring the codes . AI can generate code fast, but it still needs humans for logic, architecture, decision-making, and maintaining systems.The future is probably not **AI vs humans** it’s more like “humans who use AI” replacing those who don’t.

**Comment by BiTech5** [comment]
in 2026 most of my coding time is spent reviewing and refactoring AI-generated code instead of writing everything manually. AI is improving fast, but it still makes mistakes in production, so engineers are still needed for validation and architecture decisions. I think future developer roles may focus more on AI code reviewing, debugging, and supervising AI-generated systems rather than traditional coding alone.
In my opinion we should use AI as a tool instead of treating it as a replacement for programmers  Just like calculators never replaced mathematicians, i don't think that AI would fully replace developers

**Comment by wahabkhan000** [comment]
> I'm intrigued by the future of coding in the age of AI. With rapid advancements in ai, what transformations can we anticipate in the coding landscape? How will AI impact coding practices, job prospects, and the overall industry?

I think AI can replace coding not programming. Because coding just tackle only writing code but programming covers a whole path from problem analysis to solution implementation. Moreover, if we look at our most of the machines which humans are using now a days. It is programmed in a very structured and very complex way. And in next 10-15 years AI will not be so efficient to write and review code of these machines. As our whole technology is programmed in c or cpp which is too hard for AI to program. Like in space projects, Linux, embedded world, all of this need human programmer once again not coders. Because coding is totally different from programming. So I think there is no need to worry about.

**Comment by aaanikaagarwal-ui** [comment]
> I'm intrigued by the future of coding in the age of AI. With rapid advancements in ai, what transformations can we anticipate in the coding landscape? How will AI impact coding practices, job prospects, and the overall industry?

The role isn't disappearing; it's evolving into orchestration. As scaling AI becomes a priority, cost issues make it impossible to just throw a frontier model at every single problem. In the short term (3-5 years), there will be a ton of work for developers who can build hybrid architectures—routing complex reasoning to frontier models while handling standard processing with locally hosted, open-source alternatives. That kind of optimization is exactly where the industry is heading

**Reply by BobtheLantern** [reply]
Also, someone needs to write the prompt to answer github questions, right?

**Comment by Ahmad12-hu** [comment]
L'IA code de plus en plus à notre place, mais attention :
Comprendre les bases reste ESSENTIEL. L'IA peut se planter, faut savoir le voir et corriger.
Le vrai skill aujourd'hui : pas "demander" à l'IA, mais savoir juger ce qu'elle sort.
Continuez les fondamentaux (variables, boucles, debug). Utilisez l'IA comme assistant, pas comme béquille.

**Comment by behnoud-souhkoh** [comment]
In the future, programmers will still exist.

Yes, nowadays a normal person can build an app with ChatGPT or Claude Code, but there are still many differences between a normal person and an engineer who understands how algorithms work or how to build an API.
It’s like taking photos: everyone can take a picture now, but when it comes to professional photography, people still go to photographers.

**Reply by SjjThaler** [reply]
Mhh but will there still be such a huge demand for programmers as before. And what about people like me who are just beginning their programming journey.

**Comment by hobovsky** [comment]
_This discussion has been temporarily locked due to ongoing spam/scam messages._
